src/vm/types/serialization.rs

```diff
@@ -51,6 +51,10 @@ impl error::Error for SerializationError {
     }
 }
 
+/*
+ * jude -- it would be better for the transaction-parsing logic to know, through the type system,
+ * whether or not a failure to parse a Clarity value was due to an unexpected EOF or
+ * semantically-invalid data.  Disabling this From impl fixes this.
```

Comment:
An UnexpectedEof in the deserialization path is semantically invalid data, e.g., a byte string like `01` which indicates that it should be a `int`, but then provides no integer data would cause an unexpected EOF. However, this doesn't really affect much, since these serialization errors will cause a parse to fail in any case.